spdx/parsers/tagvaluebuilders.py

```diff
@@ -1036,66 +1112,69 @@ def set_file_name(self, doc, name):
-            # FIXME: this state does not make sense
-            self.reset_file_stat()
-            return True
-        else:
+        if not self.has_package(doc):
             raise OrderError("File::Name")
 
+        doc.packages[-1].files.append(file.File(name))
+        # A file name marks the start of a new file instance.
+        # The builder must be reset
+        # FIXME: this state does not make sense
+        self.reset_file_stat()
+        return True
+
     def set_file_spdx_id(self, doc, spdx_id):
         """
         Set the file SPDX Identifier.
         Raise OrderError if no package or no file defined.
         Raise SPDXValueError if malformed value.
         Raise CardinalityError if more than one spdx_id set.
         """
-        if self.has_package(doc) and self.has_file(doc):
-            if not self.file_spdx_id_set:
-                self.file_spdx_id_set = True
-                if validations.validate_file_spdx_id(spdx_id):
-                    self.file(doc).spdx_id = spdx_id
-                    return True
-                else:
-                    raise SPDXValueError("File::SPDXID")
-            else:
-                raise CardinalityError("File::SPDXID")
-        else:
+        if not self.has_package(doc) or not self.has_file(doc):
             raise OrderError("File::SPDXID")
 
+        if self.file_spdx_id_set:
+            raise CardinalityError("File::SPDXID")
+
+        if not validations.validate_file_spdx_id(spdx_id):
+            raise SPDXValueError("File::SPDXID")
+
+        self.file_spdx_id_set = True
+        self.file(doc).spdx_id = spdx_id
+        return True
+
     def set_file_comment(self, doc, text):
         """
         Raise OrderError if no package or no file defined.
         Raise CardinalityError if more than one comment set.
         Raise SPDXValueError if text is not free form text.
         """
-        if self.has_package(doc) and self.has_file(doc):
-            if not self.file_comment_set:
-                self.file_comment_set = True
-                if validations.validate_file_comment(text):
-                    self.file(doc).comment = str_from_text(text)
-                    return True
-                else:
-                    raise SPDXValueError("File::Comment")
-            else:
-                raise CardinalityError("File::Comment")
-        else:
+        if not self.has_package(doc) or not self.has_file(doc):
             raise OrderError("File::Comment")
 
+        if self.file_comment_set:
+            raise CardinalityError("File::Comment")
+
+        if not validations.validate_file_comment(text):
+            raise SPDXValueError("File::Comment")
+
+        self.file_comment_set = True
+        self.file(doc).comment = str_from_text(text)
+        return True
+
     def set_file_attribution_text(self, doc, text):
         """
         Set the file's attribution text .
+        Raise OrderError if no package or no file defined.
```

Comment:
Good catch!

Reply:
There might be a few more of these, but this one was glaringly obvious